offload-build.in

```diff
@@ -108,7 +108,7 @@ mapfile -t files < <(
                 printf "%s\n" "" "-> build complete" &&
                 printf "\t%s\n" "$temp"/*
             } >&2 &&
-            makepkg --packagelist
+            makepkg --config @pkgdatadir@/makepkg-x86_64.conf --packagelist
```

Comment:
This could be makepkg-i686.conf if `--arch i686` was passed, and generally would need to use the one configured with `--arch` in case we ever grow support for e.g. some ARM variant.

In commit 10c6efc440c732b70c1f11302777987cdf7e31de, support was added for e.g. installing a custom /usr/share/devtools/makepkg-myrepo-x86_64.conf, which I could use on a custom build server that provides /usr/bin/myrepo-x86_64-build. This doesn't care what the official repository PKGEXT is, so I might want to deviate... if I run `offload-build -r myrepo`, this should be expected to work. archbuild checks both in turn, and uses `makepkg-$arch.conf` as a fallback (should always be the case on mainline archlinux for extra/testing/staging/multilib).

If I locally build devtools configured for /usr/local (maybe because it is more convenient to test changes as such, without colliding with the repository package, and because unpackaged files belong in /usr/local), then try to use offload-build, it will try to find /usr/local/share/devtools/makepkg-x86_64.conf on dragon.archlinux.org, and fail, because there is a mismatch between the build `PREFIX=` between the local machine and the remote build server.

...

All these problems would not be an issue if the remote server was responsible for ensuring its /etc/makepkg.conf reported the PKGEXT which it (as a build server) is expected to produce.

Reply:
`/etc/makepkg.conf` shouldn't be the source for any crucial setting, we want to isolate them inside the devtools provided configs, except directory or other non relevant configuration for the consistency of the build env. This adds in-transparent knowledge requirement to synchronize PKGEXT in `/etc` with the one used the building the packages, which i don't like to have as a constraint.
Doing the file existence check similar to `archbuild` sounds like an appropriate thing to do, the `PREFIX` problem would however still exist. A weird way could be to check for `@pkgdatadir@/makepkg-${repo}-${arch}.conf` but use `/usr/share/devtools/makepkg-${arch}.conf"` :man_shrugging: